node/src/p2p.rs

```diff
@@ -962,6 +968,25 @@ where
         }
     }
 
+    #[instrument(level = "trace", skip(self))]
+    async fn on_header_ex_event(&mut self, ev: header_ex::Event) {
+        match ev {
+            header_ex::Event::SchedulePendingRequests => {
+                let ctx = self.swarm.context();
+
+                ctx.behaviour
+                    .header_ex
+                    .schedule_pending_requests(ctx.peer_tracker);
+            }
+            header_ex::Event::NeedTrustedPeers => {
+                self.swarm.connect_to_bootnodes();
```

Comment:
this will actually skip peers which were manually set as trusted, as we have such api